modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -226,10 +229,33 @@ const FrontendDataSet = ({
 	const isMounted = useIsMounted();
 
 	function updateDataSetItems(dataSetData) {
-		setTotal(dataSetData.totalCount);
+		setCollectionActions(dataSetData.actions);
 		setItems(dataSetData.items);
+		setTotal(dataSetData.totalCount);
 	}
 
+	useEffect(() => {
+		const filteredCreationMenu = {};
+
+		if (creationMenu && collectionActions) {
+			if (creationMenu.primaryItems) {
+				filteredCreationMenu.primaryItems = filterCreationActions(
+					creationMenu.primaryItems,
+					collectionActions
+				);
+			}
+
+			if (creationMenu.secondaryItems) {
+				filteredCreationMenu.secondaryItems = filterCreationActions(
+					creationMenu.secondaryItems,
+					collectionActions
+				);
+			}
+		}
+
+		setCreationMenuItems(filteredCreationMenu);
+	}, [creationMenu, collectionActions]);
```

Comment:
The problem is that we have `creationMenu` options because we retrieve & return them in the Fragment. However, for the `collectionActions` we need to wait for the API to request & receive the data. I think that we cannot move the logic to the `useState` initial value.

I've tried to move it to the [updateDataSetItems](https://github.com/liferay-frontend/liferay-portal/pull/3781/files#diff-8bc70260aeeec91969d2ac7da9174557ebb931b6f352d71b4e4c066e2e6449abR231). It works but then I've got unexpected `react-hooks/exhaustive-deps` errors

Reply:
> However, for the collectionActions we need to wait for the API to request & receive the data

That's not right, is it? :thinking: We should be getting them from `FrontendDataSet` API, [the creationMenu prop](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L53). In this PR, you are setting it in fragment backend:
```java
).put(
	"creationMenu",
	_getCreationActionsJSONObject(
		fdsViewObjectDefinition, fdsViewObjectEntry)
```
This is fixed, it doesn't change with returned data.

Reply:
`creationMenu` is a fix value, whatever the user defines in the DSM Actions tab. However, if I understand correctly, we need to verify for each of this actions with a `permissionKey` that it could be displayed in this menu. To to that, we need the global actions available from the API response.

![image](https://github.com/liferay-frontend/liferay-portal/assets/132653342/f36415fa-89de-4987-a27f-619b1975cd54)